### ISSUE TITLE
Use "global re" instead of importing inside methods.

### ISSUE THREAD

**Issue by NickHastings** [ganga-devs/ganga#1654]
While testing the python3 updates in the GangaND280 (see https://github.com/ganga-devs/ganga/pull/1653#issue-408228505 ) I found the Batch.submit() and Batch.resbumit() methods did not see the re module. The following error was issued:

```
"UnboundLocalError: local variable 're' referenced before assignment".
```
In two other places where re is used, the previous line has an additional "import re". So it looks like someone else encountered the same problem in other parts of Batch.py. A bit of reading indicated to me that the correct fix is to add "global re", but I confess that I didn't take the time to understand why.

So, in summary, this seems to fix a problem that I don't really understand. Anyone able to shed some light?

I've made a branch with my fix implemented and made a pull request https://github.com/ganga-devs/ganga/pull/1655#issue-408286503
